connectors/sources/graphql.py

```diff
@@ -0,0 +1,519 @@
+                "display": "dropdown",
+                "label": "Authentication Method",
+                "options": [
+                    {"label": "No Auth", "value": "none"},
+                    {"label": "Basic Auth", "value": BASIC},
+                    {"label": "Bearer Token", "value": BEARER},
+                ],
+                "order": 3,
+                "type": "str",
+                "value": "none",
+            },
+            "username": {
+                "depends_on": [{"field": "authentication_method", "value": BASIC}],
+                "label": "Username",
+                "order": 4,
+                "type": "str",
+            },
+            "password": {
+                "depends_on": [{"field": "authentication_method", "value": BASIC}],
+                "label": "Password",
+                "order": 5,
+                "sensitive": True,
+                "type": "str",
+            },
+            "token": {
+                "depends_on": [{"field": "authentication_method", "value": BEARER}],
+                "label": "Bearer Token",
+                "order": 6,
+                "sensitive": True,
+                "type": "str",
+            },
+            "graphql_query": {
+                "display": "textarea",
+                "label": "GraphQL Body",
+                "order": 7,
+                "type": "str",
+            },
+            "graphql_variables": {
+                "depends_on": [{"field": "http_method", "value": POST}],
+                "display": "textarea",
+                "label": "Graphql Variables",
+                "order": 8,
+                "type": "str",
+                "required": False,
+            },
+            "graphql_object_list": {
+                "label": "GraphQL Objects List",
+                "order": 9,
+                "tooltip": "Specifies which GraphQL objects should be indexed as individual documents. This allows finer control over indexing, ensuring only relevant data sections from the GraphQL response are stored as separate documents. Use '.' to provide full path of the object from the root of the response. For example 'organization.users.nodes'",
+                "type": "list",
+            },
+            "headers": {
+                "label": "Headers",
+                "order": 10,
+                "type": "str",
+                "required": False,
+            },
+            "pagination_model": {
+                "display": "dropdown",
+                "label": "Pagination model",
+                "options": [
+                    {"label": "No pagination", "value": NO_PAGINATION},
+                    {"label": "Cursor-based pagination", "value": CURSOR_PAGINATION},
+                ],
+                "order": 11,
+                "tooltip": "Cursor based pagination requires 'pageInfo' field along with argument 'after' for objects mentioned in 'GraphQL Objects List'. It also requires variable for 'after' argument.",
```

Comment:
`It also requires variable for 'after' argument.` - you mean we need variable within the query not a variable in the `Graphql variables` section, it got me slightly confused? I was able to perform a paginated sync without specifying any variables in `Graphql variables` - just added a variable in a query? 

We do need to add example queries in docs as configuration is pretty tricky due to the nature of graphql

<img width="629" alt="Screenshot 2024-04-02 at 10 13 05" src="https://github.com/elastic/connectors/assets/14121688/9f35f792-a737-4362-b0ef-62c032e6594d">

Reply:
Maybe something like, what do you think? I think this tooltip refers to the state before pagination key
```
For cursor-based pagination, add 'pageInfo' and an 'after' argument variable in your query at the desired node (Pagination key). Use 'after' query argument with a variable to iterate through pages. Detailed examples and setup instructions are available in the docs.
```